File: problems/917-reverse-only-letters/solution.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        letters = [c for c in s if c.isalpha()]
        letters.reverse()
        result = ""
        idx = 0
        for c in s:
            if c.isalpha():
                result += letters[idx]
                idx += 1
            else:
                result += c
        return result

File: problems/917-reverse-only-letters/solution.c
char* reverseOnlyLetters(char* s) {
    int left = 0;
    int right = strlen(s) - 1;
    while (left < right) {
        while (left < right && !isalpha(s[left])) {
            left++;
        }
        while (left < right && !isalpha(s[right])) {
            right--;
        }
        if (left < right) {
            char temp = s[left];
            s[left] = s[right];
            s[right] = temp;
            left++;
            right--;
        }
    }
    return s;
}